app/controllers/webhooks_controller.rb

```diff
@@ -69,7 +69,11 @@ def email_params
   end
 
   def authorized?
-    params[:auth].present? && params[:auth].secure_compare(ENV['WEBHOOK_AUTH_KEY'])
+    params[:auth].present? &&
+      ActiveSupport::SecurityUtils.secure_compare(
+        params[:auth],
```

Comment:
Good catch. That implementation would also leave us open to inexact matches where `WEBHOOK_AUTH_KEY` is shorter than the `auth` param, wouldn't it?

That said, the first line of `secure_compare` does this, which I'd have thought covers us (or is `bytesize` comparison not fixed time?):

```ruby
return false unless a.bytesize == b.bytesize
```